couler/core/syntax/loop.py

```diff
@@ -72,17 +83,25 @@ def map(function, input_list):
 
     inner_step.arguments = {"parameters": parameters}
 
+    # Adds values to parameters (goes through the *other with two loops.)
+    # 1. loop index of the
     with_items = []
-    for para_values in input_list:
-        item = {}
-        if not isinstance(para_values, list):
-            para_values = [para_values]
+    y = 0
+
+    while y < len(other[0]):
+
+        z = 0
+        while z < len(other):
+            item = {}
+            if not isinstance(other[z], list):
```

Comment:
The code here is a bit hard to read. Is there a way to make it look better and perhaps add some code comments on what you are doing here?

Reply:
Added a propper explanation.